packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-self-reg/java/integrationsteps.md

```diff
@@ -1,3 +1,251 @@
+   boolean canSkip = authenticationWrapper.isSkipAuthenticatorPresent(response.getProceedContext());
+   ModelAndView modelAndView = new ModelAndView("select-authenticator");
+   modelAndView.addObject("canSkip", canSkip);
+   List<String> factorMethods = new ArrayList<>();
+   for (Authenticator authenticator : response.getAuthenticators()) {
+      for (Authenticator.Factor factor : authenticator.getFactors()) {
+            factorMethods.add(factor.getMethod());
+      }
+   }
+   session.setAttribute("authenticators", response.getAuthenticators());
+   modelAndView.addObject("factorList", factorMethods);
+   modelAndView.addObject("authenticators", response.getAuthenticators());
+   modelAndView.addObject("title", title);
+   return modelAndView;
+}
+```
+
+### Step 4: Enroll required authenticator
+
+Pass the user-selected authenticator (in this case, the password authenticator) to the `IDXAuthenticationWrapper.selectAuthenticator` method.
+
+```java
+authenticationResponse = idxAuthenticationWrapper.selectAuthenticator(proceedContext, authenticator);
+```
+
+This request returns an `AuthenticationResponse` object with `AuthenticationStatus=AWAITING_AUTHENTICATOR_VERIFICATION`. You need to build a form for the user to enter their password in this authenticator verification step.
+
+<div class="common-image-format">
+
+![Displays the Java SDK password authenticator verification form](/img/oie-embedded-sdk/oie-embedded-sdk-use-case-simple-self-serv-screen-confirm-password-java.png)
+
+</div>
+
+### Step 5: Verify authenticator and display additional factors
+
+After the user enters their new password, call the `IDXAuthenticationWrapper.verifyAuthenticator` method with the user's password.
+
+```java
+VerifyAuthenticatorOptions verifyAuthenticatorOptions = new VerifyAuthenticatorOptions(newPassword);
+AuthenticationResponse authenticationResponse = idxAuthenticationWrapper.verifyAuthenticator(proceedContext, verifyAuthenticatorOptions);
+```
+
+The request returns an `AuthenticationResponse` object with `AuthenticationStatus=AWAITING_AUTHENTICATOR_ENROLLMENT_SELECTION` and an `Authenticators` list containing the email and phone factors. Reuse the authenticator enrollment form from [Step 3: Display enrollment factors](#step-3-display-enrollment-factors) to display the list of authenticators to the user.
+
+<div class="common-image-format">
+
+![Displays the Java SDK Email and Phone authenticator enrollment form](/img/oie-embedded-sdk/oie-embedded-sdk-use-case-simple-self-serv-screen-auth-list-email-phone-java.png)
+
+</div>
+
+### Step 6: User selects email authenticator
+
+In this use case, the user selects the **Email** factor as the authenticator to verify. Pass this user-selected authenticator to the `IDXAuthenticationWrapper.selectAuthenticator` method.
+
+```java
+authenticationResponse = idxAuthenticationWrapper.selectAuthenticator(proceedContext, authenticator);
```

Comment:
When the password was selected for verification, it was passed an object called `verifyAuthenticatorOptions`:

```
VerifyAuthenticatorOptions verifyAuthenticatorOptions = new VerifyAuthenticatorOptions(newPassword);
```

Shouldn't this instance use the same preceding object creation line and variable name for consistency, except with `newEmail` instead of `newPassword`?

Reply:
It's used below in step 7 too

Reply:
@vanngo-okta ?

Reply:
Sorry, I don't understand this comment.
All the code snippets, I'm taking directly from the sample app. I do not manually modify it.